### PR TITLE
Add live tests

### PR DESCRIPTION
**Pull request by kristapratico**
Test real calls to the APIs through the client using azure and openai endpoints

### CODE REVIEW

File: openai/tests/test_client.py [hunk 1]
```diff
@@ -1,7 +1,52 @@
...
+                auth="azuredefault",
+                api_version=API_VERSION,
+                backend="azure"
+            )
+        elif api_type == "openai":
+            client = openai.client.OpenAIClient(
+                auth=OPENAI_API_KEY,
+                backend="openai"
+            )
+        kwargs = {"client": client}
```
Comment (johanste) [inline openai/tests/test_client.py]: We are silently dropping any `kwargs` here. Is that intentional? If so, I'd explicitly pass in `client` to the wrapped `f` on the next line and not do a assign-new-value-to-`kwargs`-and-spread-that-in.

File: openai/tests/test_client.py [hunk 2]
```diff
@@ -10,11 +55,36 @@ def clear_oai_module(monkeypatch: pytest.MonkeyPatch):
...
     monkeypatch.setattr(openai, 'api_version', None)
+    monkeypatch.setattr(openai, 'organization', None)
 
 def setup_oai_module(monkeypatch: pytest.MonkeyPatch, **kwargs):
     for n, v in kwargs.items():
         monkeypatch.setattr(openai, n, v)
-        
+
+@pytest.fixture
+def embedding_response():
```
Comment (johanste) [inline openai/tests/test_client.py]: I think we can get rid of this fixture. I was (ab)using it incorrectly in the `assert`s. There is no reason for the mocked response to adhere to the actual output, so we can just echo the parameters instead...

File: openai/tests/test_client.py [hunk 3]
```diff
@@ -123,8 +172,320 @@ def mock_embeddings_response(*args, **kwargs):
...
+    with pytest.raises(TypeError):
+        client._normalize_model(kwargs)
+
+
+# LIVE TESTS ------------------------------------------------
+# COMPLETION TESTS
+@pytest.mark.parametrize("api_type", API_TYPE)
+@configure_client
+def test_client_completion(api_type, **kwargs):
+    client = kwargs.pop("client")
```
Comment (johanste) [inline openai/tests/test_client.py]: Why not make `client` a keyword-only argument to the test

File: openai/tests/test_client.py (hunk 2 above)
Comment (kristapratico) [reply openai/tests/test_client.py]: done

File: openai/tests/test_client.py (hunk 1 above)
Comment (kristapratico) [reply openai/tests/test_client.py]: Simplified it. Didn't realize I could pass the values of parametrize to a fixture, but pytest lets you do it cleanly.

File: openai/tests/test_client.py (hunk 3 above)
Comment (kristapratico) [reply openai/tests/test_client.py]: Was a workaround, but instead I just made client a fixture now 😆

**Review by johanste — APPROVED** [review]

**Pull request merged**